test/Atmos/Parameterizations/AerosolActivation/runtests.jl

```diff
@@ -0,0 +1,368 @@
+    param_set::EPS,
+    am::aerosol_model,
+    temp::Float64,
+    updraft_velocity::Float64,
+    G_diff::Float64,
+    press::Float64,
+)
+    mean_hygro = tp_mean_hygroscopicity(param_set, am)
+    return ntuple(am.N) do i
+        mode_i = am.modes[i]
+        num_of_comp = mode_i.n_components
+        a = sum(num_of_comp) do j
+            f = 0.5 * exp(2.5 * log(mode_i.stdev)^2)
+            g = 1 + 0.25 * log(mode_i.stdev)
+            coeff_of_curve = tp_coeff_of_curve(param_set, temp)
+            surface_tension_effects = zeta(
+                param_set,
+                temp,
+                mode_i.molar_mass[j],
+                updraft_velocity,
+                G_diff,
+            )
+            critsat =
+                2 / sqrt(mean_hygro[i]) *
+                (coeff_of_curve / (3 * mode_i.r_dry))^(3 / 2) # FILL
+
+            eta_value = eta(
+                param_set,
+                temp,
+                mode_i.molar_mass[j],
+                mode_i.N,
+                G_diff,
+                updraft_velocity,
+                press,
+            )
+
+            1 / (critsat^2) * (
+                f * (surface_tension_effects / eta_value)^(3 / 2) +
+                g * (critsat^2) /
+                (eta_value + 3 * surface_tension_effects)^(3 / 4)
+            )
+
+        end
+        a^(1 / 2)
+
+    end
+end
+
+function tp_critical_supersaturation(
+    param_set::EPS,
+    am::aerosol_model,
+    temp::Float64,
+)
+    mean_hygro = tp_mean_hygroscopicity(param_set, am)
+    return ntuple(am.N) do i
+        mode_i = am.modes[i]
+        num_of_comp = mode_i.n_components
+        a = sum(num_of_comp) do j
+            2 / sqrt(mean_hygro[i]) *
+            (tp_coeff_of_curve(param_set, temp) / (3 * mode_i.r_dry))^(3 / 2)
+        end
+        a
+    end
+
+end
+
+function tp_total_n_act(
+    param_set::EPS,
+    am::aerosol_model,
+    temp::Float64,
+    updraft_velocity::Float64,
+    G_diff::Float64,
+    press::Float64,
+)
+    critical_supersaturation = tp_critical_supersaturation(param_set, am, temp)
+    max_supersat =
+        tp_max_super_sat(param_set, am, temp, updraft_velocity, G_diff, press)
+    values = ntuple(am.N) do i
+        mode_i = am.modes[i]
+
+        sigma = mode_i.stdev
+        u_top = 2 * log(critical_supersaturation[i] / max_supersat[i])
+        u_bottom = 3 * sqrt(2) * log(sigma)
+        u = u_top / u_bottom
+        mode_i.N * 1 / 2 * (1 - erf(u))
+    end
+    summation = 0.0
+    for i in range(1, length = length(values))
+        summation += values[i]
+    end
+    return summation
+end
+
+
+
+@testset "mean_hygroscopicity" begin
+
+    println("----------")
+    println("mean_hygroscopicity: ")
+    println(tp_mean_hygroscopicity(param_set, AM_1))
+    println(mean_hygroscopicity(param_set, AM_1))
+
+    for AM in AM_test_cases
+        @test all(
+            tp_mean_hygroscopicity(param_set, AM) .≈
+            mean_hygroscopicity(param_set, AM),
```

Comment:
It looks like these tests involve copies of the functions in `tests/` and comparing against `src/`. Is there a better way we can test this?

Reply:
We are working on some sanity checks and tests comparing with the original paper. 

The functions in tests were implemented independently from the source. But I agree we need better tests than just that.